pkg/local-gadget-manager/local-gadget-manager.go

```diff
@@ -0,0 +1,309 @@
+// Copyright 2019-2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package localgadgetmanager
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	gadgetv1alpha1 "github.com/kinvolk/inspektor-gadget/pkg/api/v1alpha1"
+	"github.com/kinvolk/inspektor-gadget/pkg/container-collection"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadget-collection"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets"
+	pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/containerutils"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/stream"
+	"github.com/kinvolk/inspektor-gadget/pkg/runcfanotify"
+
+	dockertypes "github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/client"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type LocalGadgetManager struct {
+	containercollection.ContainerCollection
+
+	traceFactories map[string]gadgets.TraceFactory
+
+	// tracers by name
+	tracers map[string]tracer
+
+	runcNotifier *runcfanotify.RuncNotifier
+
+	dockercli *client.Client
```

Comment:
I suppose we want to extend this to support other container engines in the future. What do think about defining that interface right now and hiding the docker details from the localmanager?

Reply:
While trying to refactor this code, I hit a dependency problem. #312 is trying to clean up some dependency problems.